src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -93,6 +100,76 @@ class GolangSubsystem(Subsystem):
         advanced=True,
     )
 
+    cgo_enabled = BoolOption(
+        "--cgo-enabled",
+        default=False,
+        help=softwrap(
+            """\
+            Enable Cgo, which allows calling C code from Go.
+
+            See https://go.dev/blog/cgo and https://pkg.go.dev/cmd/cgo for additional information.
+            """
+        ),
+    )
+
+    _cgo_tool_search_paths = StrListOption(
+        default=["<PATH>"],
+        help=softwrap(
+            """
+            A list of paths to search for tools needed by CGo (e.g., gcc, g++).
+
+            Specify absolute paths to directories with tools needed by CGo , e.g. `/usr/bin`.
+            Earlier entries will be searched first.
+
+            The following special strings are supported:
+
+              * `<PATH>`, the contents of the PATH environment variable
+            """
+        ),
+    )
+
+    cgo_gcc_binary_name = StrOption(
+        default="gcc",
+        advanced=True,
+        help="Name of the tool to use to compile C code included via CGo in a Go package.",
+    )
+
+    cgo_gxx_binary_name = StrOption(
+        default="g++",
+        advanced=True,
+        help="Name of the tool to use to compile C++ code included via CGo in a Go package.",
+    )
+
+    cgo_fortran_binary_name = StrOption(
+        default="gfortran",
+        advanced=True,
+        help="Name of the tool to use to compile fortran code included via CGo in a Go package.",
+    )
+
+    cgo_default_cflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
+        advanced=True,
+        help="Default options used for CFLAGS when compiling cgo code. Equivalent to CGO_CFLAGS passed to `go`.",
+    )
+
+    cgo_default_cxxflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
+        advanced=True,
+        help="Default options used for CXXFLAGS when compiling cgo code. Equivalent to CGO_CXXFLAGS passed to `go`.",
+    )
+
+    cgo_default_fflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
+        advanced=True,
+        help="Default options used for FFLAGS when compiling cgo code. Equivalent to CGO_FFLAGS passed to `go`.",
+    )
+
+    cgo_default_ldflags = StrListOption(
+        default=lambda _: list(_DEFAULT_COMPILER_FLAGS),
```

Comment:
Is this a correct default?

Reply:
I believe so given the `go` source here: https://github.com/golang/go/blob/202b7e7e76128f524142ac7d328fe8458a096dbf/src/cmd/go/internal/work/exec.go#L2724

I would like a second opinion though on the defaults for all of those variables being `-g `-O2`.

Reply:
Ah, okay, I'm totally fine passing through the `cgo` defaults following my principle-of-least-surprise methodology. 

I was a bit surprised at `-g` being passed through to `ld` as I thought `ld` ignores it by default, but I guess it's harmless enough.

Reply:
Also, my opinion is predominantly embedded, where I almost exclusively use `-Og` and `-Ofast`, but I wouldn't propose those here.